Prepare release notes for v0.20.0 (#6556)
scikit-image/scikit-image: doc/release/release_0.20.rst

Announcement: scikit-image 0.20.0
=================================

We're happy to announce the release of scikit-image v0.20.0!

> scikit-image is an image processing toolbox for SciPy that includes algorithms

Reviewer: ```suggestion
scikit-image is an image processing toolbox built on SciPy that includes algorithms
```

for segmentation, geometric transformations, color space manipulation,
analysis, filtering, morphology, feature detection, and more.

For more information, examples, and documentation, please visit our website:

https://scikit-image.org

New features and improvements
-----------------------------
- Several footprint generating and consuming functions in ``skimage.morphology`` now support footprint decomposition.

>   By decomposing a footprint into several smaller ones, morphological operations can potentially be speed up.

Reviewer 2: ```suggestion
  By decomposing a footprint into several smaller ones, morphological operations can potentially be sped up.
```

>   The decomposed footprint can be generated with the new ``decomposition`` parameter of the functions ``rectangle``, ``diamond``, ``disk``, ``cube``, ``octahedron``, ``ball``, ``octagon`` in ``skimage.morphology``.

Reviewer: ```suggestion
  The decomposed footprint can be generated with the new ``decomposition`` parameter of the functions ``rectangle``, ``diamond``, ``disk``, ``cube``, ``octahedron``, ``ball``, and ``octagon`` in ``skimage.morphology``.
```

>   The ``footprint`` parameter of the functions ``binary_erosion``, ``binary_dilation``, ``binary_opening``, ``binary_closing``, ``erosion``, ``dilation``, ``opening``, ``closing``, ``white_tophat``, and ``black_tophat`` in ``skimage.morphology`` now accepts a tuple of a footprint and the number of times it should be applied iteratively.

Reviewer: Unclear: what does tuple of a footprint mean, and how would this be used?

Author: I'll try to clarify. :+1:

Reviewer 3: How about 

"now accepts a sequence of 2-tuples ``(footprint_i, num_iter_i)`` where each entry, ``i``, of the sequence contains a footprint and the number of times it should be iteratively applied. This is the form produced by the footprint decompositions mentioned above."

Author: Sounds good. Done.

  (`#5482 <https://github.com/scikit-image/scikit-image/pull/5482>`_, `#6151 <https://github.com/scikit-image/scikit-image/pull/6151>`_)

> - Added support for the Modified Hausdorff Distance (MHD) metric in ``skimage.metrics.hausdorff_distance`` which performs better than the directed Hausdorff Distance (HD).

Reviewer: What does "behave better" here?

Author: This refers to this statement in #5581

> Dubuisson et al. [1] demonstrated that the Modified Hausdorff Distance (MHD) outperforms the directed Hausdorff Distance (HD).

 From the [paper](https://doi.org/10.1109/ICPR.1994.576361)

> We have determined that, among the class of distance measures based on the Hausdorff distance, the
modified Hausdorff distance (MHD) is best for matching two objects based on their edge points. We have
shown that MHD has the following desirable properties: (i) its value increases monotonically as the
amount of difference between the two sets of edge points increases, and (ii) it is robust t o outlier points
that might result from segmentation errors.

My intention is that the notes answer "why did we add this" briefly. I'm trying to keep a good balance between not digging into every PR to deeply and still being clear. I'll try to clarify the advantages here.

> - Added support for the Modified Hausdorff Distance (MHD) metric in ``skimage.metrics.hausdorff_distance`` which performs better than the directed Hausdorff Distance (HD).

Reviewer: ```suggestion
- Add support for the Modified Hausdorff Distance (MHD) metric in ``skimage.metrics.hausdorff_distance`` which performs better than the directed Hausdorff Distance (HD).
```

Author: I'd argue these notes are read after the release. And for [past releases](https://scikit-image.org/docs/0.19.x/release_notes.html#new-features) we seem to have used past tense as well?

Reviewer: I've always considered it standard practice to use present tense for changelogs. I haven't been making releases for a while, but you'll notice this in older release notes:

https://scikit-image.org/docs/0.14.x/release_notes_and_installation.html

This derives from *commit messages* being written in the imperative mood. See, e.g.:

https://vxlabs.com/software-development-handbook/#good-commit-messages

  (`#5581 <https://github.com/scikit-image/scikit-image/pull/5581>`_)

> - Added support for the Modified Hausdorff Distance (MHD) metric in ``skimage.metrics.hausdorff_distance`` via the new parameter ``method`` .

Reviewer: ```suggestion
- Add support for the Modified Hausdorff Distance (MHD) metric in ``skimage.metrics.hausdorff_distance`` via the new parameter ``method`` .
```

  (`#5581 <https://github.com/scikit-image/scikit-image/pull/5581>`_)

> - Added two datasets ``skimage.data.protein_transport`` and ``skimage.data.nickel_solidification``. (`#6087 <https://github.com/scikit-image/scikit-image/pull/6087>`_)

Reviewer 2: ```suggestion
- Added two datasets ``skimage.data.protein_transport`` and ``skimage.data.nickel_solidification`` (`#6087 <https://github.com/scikit-image/scikit-image/pull/6087>`_).
```

Author: On the danger of bikeshedding, I'm emulating NumPy here, e.g. [v1.23.0](https://github.com/numpy/numpy/releases/tag/v1.23.0). And I would argue that the referenced PRs are not part of the last sentence but rather an addendum of the whole bullet point.

> - Added two datasets ``skimage.data.protein_transport`` and ``skimage.data.nickel_solidification``. (`#6087 <https://github.com/scikit-image/scikit-image/pull/6087>`_)

Reviewer: ```suggestion
- Add two datasets ``skimage.data.protein_transport`` and ``skimage.data.nickel_solidification``. (`#6087 <https://github.com/scikit-image/scikit-image/pull/6087>`_)
```

Changes and new deprecations
----------------------------

Completed deprecations
----------------------

> Bugfixes

Reviewer 2: ```suggestion
Bug fixes
```

Reviewer: Remember to also update underline length

--------

Documentation
-------------

> - Add a textbook-like tutorial on measuring fluorescence at the nuclear envelope.

Reviewer 2: ```suggestion
- Add a textbook-like tutorial on measuring fluorescence at the nuclear envelope of a cell
```
(just to provide a bit more context)

>   (`#5262 <https://github.com/scikit-image/scikit-image/pull/5262>`_)

Reviewer 2: ```suggestion
  (`#5262 <https://github.com/scikit-image/scikit-image/pull/5262>`_).
```

Other and development related updates
-------------------------------------

> - Added benchmarks for ``morphology.local_maxima``. (`#3255 <https://github.com/scikit-image/scikit-image/pull/3255>`_)

Reviewer: ```suggestion
- Add benchmarks for ``morphology.local_maxima``. (`#3255 <https://github.com/scikit-image/scikit-image/pull/3255>`_)
```

- Fix the autogeneration of API docs for lazy loaded subpackages. (`#6075 <https://github.com/scikit-image/scikit-image/pull/6075>`_)
- Checkout gh-pages with a shallow clone. (`#6085 <https://github.com/scikit-image/scikit-image/pull/6085>`_)

TODO
----

> - Transplant the change of scikit-learn into scikit-image for RANSAC  (`#6046 <https://github.com/scikit-image/scikit-image/pull/6046>`_)

Reviewer: ```suggestion
- Incorporate RANSAC improvements from scikit-learn into scikit-image (`#6046 <https://github.com/scikit-image/scikit-image/pull/6046>`_)
```

Author: Thanks. I haven't gotten to these Todo items yet and will be gradually moving them over to their correct section. :+1:

- Skip tests requiring fetched data (`#6089 <https://github.com/scikit-image/scikit-image/pull/6089>`_)
- Fix dev doc build (`#6091 <https://github.com/scikit-image/scikit-image/pull/6091>`_)
- forward port of `#6098 <https://github.com/scikit-image/scikit-image/pull/6098>`_ (fix MacOS arm64 wheels and Windows Python 3.10 AMD64 wheel) (`#6101 <https://github.com/scikit-image/scikit-image/pull/6101>`_)

> - make rank filter test comparisons robust across architectures (`#6103 <https://github.com/scikit-image/scikit-image/pull/6103>`_)

Reviewer: ```suggestion
- Make rank filter test comparisons robust across architectures (`#6103 <https://github.com/scikit-image/scikit-image/pull/6103>`_)
```

> - pass a specific random_state into ransac in test_ransac_geometric (`#6105 <https://github.com/scikit-image/scikit-image/pull/6105>`_)

Reviewer: ```suggestion
- Pass a specific random_state into ransac in test_ransac_geometric (`#6105 <https://github.com/scikit-image/scikit-image/pull/6105>`_)
```

- Add linker flags to strip debug symbols during wheel building (`#6109 <https://github.com/scikit-image/scikit-image/pull/6109>`_)

> - relax test condition to make it more robust to variable CI load (`#6114 <https://github.com/scikit-image/scikit-image/pull/6114>`_)

Reviewer: ```suggestion
- Relax test condition to make it more robust to variable CI load (`#6114 <https://github.com/scikit-image/scikit-image/pull/6114>`_)
```

Reviewer: Just capitalize all of these.

- Fix wrong doc on connected pixels in flood (`#6534 <https://github.com/scikit-image/scikit-image/pull/6534>`_)
- Minor doc fix: add missing print statement in the `plot_segmentations.py` example (`#6535 <https://github.com/scikit-image/scikit-image/pull/6535>`_)
- Apply codespell to fix common spelling mistakes (`#6537 <https://github.com/scikit-image/scikit-image/pull/6537>`_)
- Ignore codespell fixes with git blame (`#6539 <https://github.com/scikit-image/scikit-image/pull/6539>`_)
- Add missing spaces to regionprops error message. (`#6545 <https://github.com/scikit-image/scikit-image/pull/6545>`_)
- Update "Mark dormant issues" workflow (`#6546 <https://github.com/scikit-image/scikit-image/pull/6546>`_)
- Add missing space in math directive in normalized_mutual_information's docstring (`#6549 <https://github.com/scikit-image/scikit-image/pull/6549>`_)
- Add missing option stale-pr-label for "Mark dormant issues" workflow (`#6552 <https://github.com/scikit-image/scikit-image/pull/6552>`_)
- Remove FUNDING.yml in preference of org version (`#6553 <https://github.com/scikit-image/scikit-image/pull/6553>`_)

Pull Requests in this release
-----------------------------

> - Add benchmarks for morphology.local_maxima (`#3255 <https://github.com/scikit-image/scikit-image/pull/3255>`_)

Reviewer: We should make sure there are no duplicates

Author: I intend to fully delete the previous section "Todo". Whether we want to include this short list of all PR at the end similar to the contributor and reviewer lists is up to you.

- Minor doc fix: add missing print statement in the `plot_segmentations.py` example (`#6535 <https://github.com/scikit-image/scikit-image/pull/6535>`_)
- Apply codespell to fix common spelling mistakes (`#6537 <https://github.com/scikit-image/scikit-image/pull/6537>`_)
- Ignore codespell fixes with git blame (`#6539 <https://github.com/scikit-image/scikit-image/pull/6539>`_)
- Add missing spaces to regionprops error message. (`#6545 <https://github.com/scikit-image/scikit-image/pull/6545>`_)
- Update "Mark dormant issues" workflow (`#6546 <https://github.com/scikit-image/scikit-image/pull/6546>`_)
- Add missing space in math directive in normalized_mutual_information's docstring (`#6549 <https://github.com/scikit-image/scikit-image/pull/6549>`_)
- Add missing option stale-pr-label for "Mark dormant issues" workflow (`#6552 <https://github.com/scikit-image/scikit-image/pull/6552>`_)
- Remove FUNDING.yml in preference of org version (`#6553 <https://github.com/scikit-image/scikit-image/pull/6553>`_)

56 authors added to this release [alphabetical by first name or login]
----------------------------------------------------------------------

> - =

Reviewer: ```suggestion
```

- Adeel Hassan
- Albert Y. Shih

> - AleixBP (AleixBP)

Reviewer: ```suggestion
- AleixBP
```

- Alexandr Kalinin
- Alexandre de Siqueira
- Antony Lee
- Balint Varga
- Ben Greiner

> - bsmietanka (bsmietanka)

Reviewer: ```suggestion
- bsmietanka
```

- Chris Roat
- Chris Wood
- Dave Mellert
- Dudu Lasry
- Elena Pascal
- Fabian Schneider
- Frank A. Krueger
- Gregory Lee
- Hande Gözükan
- Jacob Rosenthal
- James Gao
- Jan Kadlec
- Jan-Hendrik Müller
- Jan-Lukas Wynen
- Jarrod Millman

> - johnthagen (johnthagen)

Reviewer: ```suggestion
- johnthagen
```

- Joshua Newton
- Juan DF
- Juan Nunez-Iglesias
- Judd Storrs
- Larry Bradley
- Lars Grüter
- Lucas Johnson
- maldil (maldil)
- Marianne Corvellec
- Mark Harfouche
- Marvin Albert
- Miles Lucas
- Naveen

> - pbuscay (Preston Buscay)

Reviewer: ```suggestion
- Preston Buscay
```

> - peterbell10 (peterbell10)

Reviewer: ```suggestion
- Peter Bell
```

- Ray Bell

> - Riadh

Reviewer: ```suggestion
```

- Riadh Fezzani
- Robin Thibaut
- Ross Barnowski
- Sandeep N Menon
- Sanghyeok Hyun
- Sebastian Wallkötter
- Simon-Martin Schröder
- Stefan van der Walt
- Teemu Kumpumäki

> - thvoigtmann (Thomas Voigtmann)

Reviewer: ```suggestion
- Thomas Voigtmann
```

- thvoigtmann (thvoigtmann)
- Tim-Oliver Buchholz
- Tyler Reddy

30 reviewers added to this release [alphabetical by first name or login]
------------------------------------------------------------------------
- Abhijeet Parida
- Albert Y. Shih
- Alexandre de Siqueira
- Antony Lee
- Ben Greiner
- Carlo
- Chris Roat
- Dudu Lasry
- François Boulogne
- Gregory Lee
- Jacob Rosenthal
- James Gao
- Jan-Hendrik Müller
- Jarrod Millman
- Juan DF
- Juan Nunez-Iglesias
- Lars Grüter
- maldil
- Marianne Corvellec
- Mark Harfouche
- Marvin Albert
- Riadh Fezzani
- Robert Haase
- Robin Thibaut
- Sandeep N Menon
- Sanghyeok Hyun
- Sebastian Wallkötter
- Stefan van der Walt

> - thvoigtmann

Reviewer: ```suggestion
- Thomas Voigtmann
```

  (`#3255 <https://github.com/scikit-image/scikit-image/pull/3255>`_).
- Fix the autogeneration of API docs for lazy loaded subpackages
  (`#6075 <https://github.com/scikit-image/scikit-image/pull/6075>`_).
- Checkout gh-pages with a shallow clone
  (`#6085 <https://github.com/scikit-image/scikit-image/pull/6085>`_).
- Fix dev doc build
  (`#6091 <https://github.com/scikit-image/scikit-image/pull/6091>`_).
- Expand reviewer guidelines in pull request template
  (`#6208 <https://github.com/scikit-image/scikit-image/pull/6208>`_).
- Move ``skimage/measure/mc_meta`` folder into ``tools/precompute/`` folder to avoid its unnecessary distribution to users
  (`#6294 <https://github.com/scikit-image/scikit-image/pull/6294>`_).
- Remove unused function ``getLutNames`` in ``tools/precompute/mc_meta/createluts.py``
  (`#6294 <https://github.com/scikit-image/scikit-image/pull/6294>`_).

TODO Milestone 1.0?
-------------------
- Fix inpaint_biharmonic for images with Fortran-ordered memory layout (`#6263 <https://github.com/scikit-image/scikit-image/pull/6263>`_)
- Support array-likes consistently in geometric transforms (`#6270 <https://github.com/scikit-image/scikit-image/pull/6270>`_)

> Backported 0.19.x
> ---------------

Reviewer 4: ```suggestion
Backported 0.19.x
-----------------
```

  (`#6257 <https://github.com/scikit-image/scikit-image/pull/6257>`_).
- Support the construction of ``skimage.io.ImageCollection`` from a ``load_pattern`` with an arbitrary sequence as long as a matching ``load_func`` is provided
  (`#6276 <https://github.com/scikit-image/scikit-image/pull/6276>`_).
- Warn for non-integer image inputs to ``skimage.feature.local_binary_pattern``.
  Applying the function to floating-point images may give unexpected results when small numerical differences between adjacent pixels are present
  (`#6272 <https://github.com/scikit-image/scikit-image/pull/6272>`_).
- Use the minimal required unsigned integer size in ``skimage.filters.rank_order`` which allows to operate the function with higher precision or on larger arrays.
  Previously, the returned ``labels`` and and ``original_values`` were always of type uint32.
  (`#6342 <https://github.com/scikit-image/scikit-image/pull/6342>`_).
- Use the minimal required unsigned integer internally in ``skimage.morphology.reconstruction`` which allows to operate the function with higher precision or on larger arrays.
  Previously, int32 was used.

>   (`#6342 <https://github.com/scikit-image/scikit-image/pull/6342>`_).

Author: @grlee77, could you please have look if I correctly summarized https://github.com/scikit-image/scikit-image/pull/6342? 

> single precision seed now returns single precision output (this function was missed during previous float32 work)

I'm confused about this bit, because I don't see any changes in the PRs diff that should impact `seed`?

>   (`#6342 <https://github.com/scikit-image/scikit-image/pull/6342>`_).

Author: @grlee77, could you please have look if I correctly summarized https://github.com/scikit-image/scikit-image/pull/6342?

> single precision seed now returns single precision output (this function was missed during previous float32 work)

I'm confused about this bit, because I don't see any changes in the PRs diff that should impact seed?

Reviewer 3: This was just referring to a place where `morphology.reconstruct` had unconditional promotion to float64 internally. See at the left in the diff here on lines 173-175 where `seed` and `mask` used to get copied into a float64 array. This was changed to use `_supported_float_type` so it is possible to get single precision output from this function.

https://github.com/scikit-image/scikit-image/pull/6342/files#diff-a0783c211f0b0a66e4cc41bf1838cdafe46ab957a9dfdb685784aecd64135243L173

Author: Thanks for the clarification. So actually `mask` determines the dtype of the output and not `seed`?

Reviewer 2: Good question... I would assume we would want "float_dtype = max { _supported_float_type(seed.dtype), _supported_float_type(mask.dtype) }" :thinking:

- Fix calculation of Z normal in marching cubes (`#6227 <https://github.com/scikit-image/scikit-image/pull/6227>`_)
- Remove redundant testing on Appveyor (`#6229 <https://github.com/scikit-image/scikit-image/pull/6229>`_)
- Update imports/refs from deprecated scipy.ndimage.filters namespace (`#6231 <https://github.com/scikit-image/scikit-image/pull/6231>`_)
- Include Cython sources via package_data (`#6232 <https://github.com/scikit-image/scikit-image/pull/6232>`_)
- DOC: fix SciPy intersphinx (`#6239 <https://github.com/scikit-image/scikit-image/pull/6239>`_)
- Fix bug in SLIC superpixels with `enforce_connectivity=True` and `start_label > 0` (`#6242 <https://github.com/scikit-image/scikit-image/pull/6242>`_)
- Fowardport PR #6249 on branch main (update MacOS libomp installation in wheel building script) (`#6250 <https://github.com/scikit-image/scikit-image/pull/6250>`_)
- Ignore sparse matrix deprecation warning (`#6261 <https://github.com/scikit-image/scikit-image/pull/6261>`_)

58 authors added to this release [alphabetical by first name or login]
----------------------------------------------------------------------

> - Adeel Hassan

Author: What do you think about shorting this list (and the one for the reviewers) to an enumeration like "Person 1, Person 2, ..." to save vertical space?

Reviewer 2: I'd be fine with this change; it would be still be easily parseable. But if it's one line, it might hurt horizontally... I admit I don't like horizontal scrolling. Maybe wrapping lines at 80 or 100 characters?

Author: I think wrapping would make parsing more difficult. I haven't payed attention to horizontal scrolling throughout the document anyway. From my perspective I don't see it as a problem (soft-wrapping editors exist), but I'm aware that not everyone may feel like this. I'm happy to leave it either way.

Reviewer 2: Maybe I'm just feeling lazy today, but since this (long) list lives at the very end of the file, I'm actually fine keeping things the way they are now...

  The new properties ``num_pixels`` and ``coords_scaled`` are available as well.
  See the respective docstrings for more details
  (`#6296 <https://github.com/scikit-image/scikit-image/pull/6296>`_).
- Add the isotropic binary morphological operators ``isotropic_closing``, ``isotropic_dilation``, ``isotropic_erosion``, and ``isotropic_opening`` in ``skimage.morphology``.
  These function return the same results as their non-isotropic counterparts but perform faster for large circular structuring elements
  (`#6492 <https://github.com/scikit-image/scikit-image/pull/6492>`_).
- Support the Modified Hausdorff Distance (MHD) metric in ``skimage.metrics.hausdorff_distance`` via the new parameter ``method``.
  The MHD can be more robust against outliers than the directed Hausdorff Distance (HD)
  (`#5581 <https://github.com/scikit-image/scikit-image/pull/5581>`_).
- Added two datasets ``skimage.data.protein_transport`` and ``skimage.data.nickel_solidification``
  (`#6087 <https://github.com/scikit-image/scikit-image/pull/6087>`_).

> - Add the new parameter ``use_gaussian_derivatives`` to ``skimage.feature.hessian_matrix`` which allows the computation of the Hessian matrix by convolving with Gaussian derivatives

Author: It feels grammatically wrong to leave out the article here. Would you prefer "Add a new parameter..." instead?

> - Add the new parameter ``use_gaussian_derivatives`` to ``skimage.feature.hessian_matrix`` which allows the computation of the Hessian matrix by convolving with Gaussian derivatives

Reviewer 2: @lagru it's like "Go to gate B." or "Film director James Cameron is famous." (you don't say: "Go to *the* gate B." nor "*The* film director James Cameron is famous.") -- I searched quickly but, unfortunately, I couldn't find a reference to point you to... With the indefinite article, it could be "Add a new parameter (namely, `use_gaussian_derivatives`) ..."

> - Add the new parameter ``use_gaussian_derivatives`` to ``skimage.feature.hessian_matrix`` which allows the computation of the Hessian matrix by convolving with Gaussian derivatives

Author: Okay, that seems convincing. Thanks for the explanation. I'm happy to go along with the "Go to gate B" version. :smile:

  (`#6149 <https://github.com/scikit-image/scikit-image/pull/6149>`_).

> - Add new parameters ``squared_butterworth`` and ``npad`` to ``skimage.filters.butterworth``, which add support for squaring the filter and edge padding

Author: Hmm, I found the "(non)" a bit strange to read. How about

> Add new parameters ``squared_butterworth`` and ``npad`` to ``skimage.filters.butterworth``, which support traditional or squared filtering and edge padding, respectively

  (`#6509 <https://github.com/scikit-image/scikit-image/pull/6509>`_).
- Use the minimal required unsigned integer internally in ``skimage.morphology.reconstruction`` which allows to operate the function with higher precision or on larger arrays.
  Previously, int32 was used.
  (`#6342 <https://github.com/scikit-image/scikit-image/pull/6342>`_).
- Use the minimal required unsigned integer size in ``skimage.filters.rank_order`` which allows to operate the function with higher precision or on larger arrays.
  Previously, the returned ``labels`` and and ``original_values`` were always of type uint32.
  (`#6342 <https://github.com/scikit-image/scikit-image/pull/6342>`_).

Changes and new deprecations
----------------------------
- Rewrite ``skimage.filters.meijering``, ``skimage.filters.sato``,
  ``skimage.filters.frangi``, and ``skimage.filters.hessian`` to match the published algorithms more closely.
  This change is backward incompatible and will lead to different output values compared to the previous implementation.
  The Hessian matrix calculation is now done more accurately.

>   The filters will now correctly be set to zero whenever one of the hessian eigenvalues has a sign which is incompatible with a ridge of the desired polarity.

Reviewer 2: ```suggestion
  The filters will now be correctly set to zero whenever one of the Hessian eigenvalues has a sign which is incompatible with a ridge of the desired polarity.
```

  The gamma constant of the Frangi filter is now set adaptively based on the maximum Hessian norm
  (`#6446 <https://github.com/scikit-image/scikit-image/pull/6446>`_).
- Return ``False`` in ``skimage.measure.LineModelND.estimate`` instead of raising an error if the model is under-determined
  (`#6453 <https://github.com/scikit-image/scikit-image/pull/6453>`_).
- Return ``False`` in ``skimage.measure.CircleModel.estimate`` instead of warning if the model is under-determined
  (`#6453 <https://github.com/scikit-image/scikit-image/pull/6453>`_).
- Rename ``skimage.filter.inverse`` to ``skimage.filter.inverse_filter``.

>   ``skimage.filter.inverse`` is deprecated and will be removed in the next release

Reviewer 2: ```suggestion
  ``skimage.filter.inverse`` is deprecated and will be removed in the next (minor) release
```

Reviewer 2: Sure, cf. related discussion we've had on HackMD :+1:

  (`#6418 <https://github.com/scikit-image/scikit-image/pull/6418>`_).
- Update minimal supported dependencies to ``numpy>=1.20``
  (`#6565 <https://github.com/scikit-image/scikit-image/pull/6565>`_).
- Update minimal supported dependencies to ``scipy>=1.8``
  (`#6564 <https://github.com/scikit-image/scikit-image/pull/6564>`_).
- Update minimal supported dependencies to ``networkx>=2.8``
  (`#6564 <https://github.com/scikit-image/scikit-image/pull/6564>`_).
- Update minimal supported dependency to ``pillow>=9.0.1``
  (`#6402 <https://github.com/scikit-image/scikit-image/pull/6402>`_).
- Update optional, minimal supported dependency to ``matplotlib>=3.3``
  (`#6383 <https://github.com/scikit-image/scikit-image/pull/6383>`_).
- Warn for non-integer image inputs to ``skimage.feature.local_binary_pattern``.
  Applying the function to floating-point images may give unexpected results when small numerical differences between adjacent pixels are present
  (`#6272 <https://github.com/scikit-image/scikit-image/pull/6272>`_).

Completed deprecations
----------------------

> - Remove ``skimage.viewer`` which was scheduled for the postponed version 1.0

Reviewer 2: Either "the removal of which" or "which was scheduled for removal on the postponed version 1.0"

  (`#6525 <https://github.com/scikit-image/scikit-image/pull/6525>`_).
- Fix unexpected error when passing a tuple to the parameter ``exclude_border`` in ``skimage.feature.blog_dog`` and ``skimage.feature.blob_log``
  (`#6533 <https://github.com/scikit-image/scikit-image/pull/6533>`_).
- Raise a specific error message in ``skimage.segmentation.random_walker`` if no seeds are provided as positive values in the parameter ``labels``
  (`#6562 <https://github.com/scikit-image/scikit-image/pull/6562>`_).
- Raise a specific error message when accessing region properties from ``skimage.measure.regionprops`` when the required  ``intensity_image`` is unavailable
  (`#6584 <https://github.com/scikit-image/scikit-image/pull/6584>`_).
- Avoid errors in ``skimage.feature.ORB.detect_and_extract`` by breaking early if the octave image is too small
  (`#6590 <https://github.com/scikit-image/scikit-image/pull/6590>`_).
- Fix ``skimage.restoration.inpaint_biharmonic`` for images with Fortran-ordered memory layout (`#6263 <https://github.com/scikit-image/scikit-image/pull/6263>`_)

Documentation
-------------
.. Everything regarding the documentation that is visible to users.

New
~~~

> - Add a textbook-like tutorial on measuring fluorescence at the nuclear envelope of a cell
>   (`#5262 <https://github.com/scikit-image/scikit-image/pull/5262>`_).

Reviewer 2: ```suggestion
```
This isn't new! It was added in v0.19.2 (see https://github.com/scikit-image/scikit-image/blob/main/doc/release/release_0.19.rst#scikit-image-0192).

> - Add new gallery example on decomposing flat footprints

Reviewer 2: ```suggestion
- Add a gallery example on decomposing flat footprints
```

  (`#6151 <https://github.com/scikit-image/scikit-image/pull/6151>`_).

> - Add a new and gallery example "Butterworth Filters" and improve docstring of ``skimage.filters.butterworth``

Reviewer 2: ```suggestion
- Add gallery example "Butterworth Filters" and improve docstring of ``skimage.filters.butterworth``
```

  (`#6251 <https://github.com/scikit-image/scikit-image/pull/6251>`_).

> - Add a new gallery example "Render text onto an image"

Reviewer 2: ```suggestion
- Add gallery example "Render text onto an image"
```

  (`#6431 <https://github.com/scikit-image/scikit-image/pull/6431>`_).

> - Add support page (``.github/SUPPORT.md``)to help users from GitHub find appropriate support resources

Reviewer 2: ```suggestion
- Add support page (``.github/SUPPORT.md``) to help users from GitHub find appropriate support resources
```

  (`#6171 <https://github.com/scikit-image/scikit-image/pull/6171>`_, `#6575 <https://github.com/scikit-image/scikit-image/pull/6575>`_).
- Add ``CITATION.bib`` to repository to help with citing scikit-image
  (`#6195 <https://github.com/scikit-image/scikit-image/pull/6195>`_).

Improved & updated
~~~~~~~~~~~~~~~~~~

> - Improve the the gallery example "Measure perimeters with different estimators"

Reviewer 2: ```suggestion
- Improve gallery example "Measure perimeters with different estimators"
```

  (`#6200 <https://github.com/scikit-image/scikit-image/pull/6200>`_, `#6121 <https://github.com/scikit-image/scikit-image/pull/6121>`_).

> - Adapt the gallery example "Build image pyramids" to more diversified shaped images and downsample factors

Reviewer 2: ```suggestion
- Adapt gallery example "Build image pyramids" to more diversified shaped images and downsample factors
```

  (`#6421 <https://github.com/scikit-image/scikit-image/pull/6421>`_, `#6422 <https://github.com/scikit-image/scikit-image/pull/6422>`_).
- Replace reference to ``api_changes.rst`` with ``release_dev.rst``
  (`#6495 <https://github.com/scikit-image/scikit-image/pull/6495>`_).
- Clarify header pointing to notes for latest version released
  (`#6508 <https://github.com/scikit-image/scikit-image/pull/6508>`_).
- Add missing spaces to error message in ``skimage.measure.regionprops``
  (`#6545 <https://github.com/scikit-image/scikit-image/pull/6545>`_).
- Apply codespell to fix common spelling mistakes
  (`#6537 <https://github.com/scikit-image/scikit-image/pull/6537>`_).
- Add missing space in math directive in normalized_mutual_information's docstring
  (`#6549 <https://github.com/scikit-image/scikit-image/pull/6549>`_).
- Fix lengths of docstring heading underline in ``skimage.morphology.isotropic_`` functions
  (`#6628 <https://github.com/scikit-image/scikit-image/pull/6628>`_).
- Fix plot order due to duplicate examples with the file name ``plot_thresholding.py``
  (`#6644 <https://github.com/scikit-image/scikit-image/pull/6644>`_).

> - Get rid of numpy dep warning in gallery example ``plot_equalize``

Reviewer 2: ```suggestion
- Get rid of numpy deprecation warning in gallery example ``plot_equalize``
```
To me "dep" means "dependency" :wink:

Author: I'm confused. You say it means "dependency" but you suggest "deprecation"? :sweat_smile: I'm guessing deprecation is the one you mean?

Reviewer 2: Nope: Here, "dep" means "deprecation" (what would a "dependency warning" would be, btw? :wink:) -- so let's spell it out. Plus, let's avoid any confusion, because 'usually' "dep" is used to mean "dependency" -- at least in my dictionary.

>   (`#6650 <https://github.com/scikit-image/scikit-image/pull/6650>`_).

Reviewer 2: ```suggestion
  (`#6650 <https://github.com/scikit-image/scikit-image/pull/6650>`_).
- Fix swapping of opening and closing in gallery example ``plot_rank_filters``
  (`#6652 <https://github.com/scikit-image/scikit-image/pull/6652>`_).
```
